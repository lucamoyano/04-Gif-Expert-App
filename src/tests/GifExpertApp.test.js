import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas en componente GifExpertApp', () => {

    test('debe mostrar correctamente correctamente', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {
        const categories = ['Dragon Ball', 'Naruto'];
        const wrapper = shallow( <GifExpertApp  defaultCategories = { categories } /> );

        expect( wrapper ).toMatchSnapshot();
        //Evaulo si tengo la misma cantidad de categorias que categories
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
    

});