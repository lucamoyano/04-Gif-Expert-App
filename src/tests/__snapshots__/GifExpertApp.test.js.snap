// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifExpertApp debe mostrar correctamente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas en componente GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;
